test/lib/test-index.ts

```diff
@@ -0,0 +1,58 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import assert = require('assert');
+import http = require('http');
+import mockery = require('mockery');
+import _ = require('lodash');
+import MockBLPAPI = require('./mock-blpapi');
+import sio = require('socket.io');
+
+// Setup mock of blpapi
+mockery.registerMock('blpapi', MockBLPAPI);
+mockery.enable({
+    useCleanCache: true,
+    warnOnReplace: true,
+    warnOnUnregistered: false
+});
+
+// Setup an Socket.IO server for IPC need
+var s: http.Server = http.createServer();
+s.listen(3333);
+s.on('listening', (): void => {
+    console.log('ipc server listening on port 3333');
+});
+var io: SocketIO.Server = sio(s);
+io.on('connection', (socket: SocketIO.Socket): void => {
+    console.log('ipc channel connected');
+    MockBLPAPI.ipc = socket;
+    socket.on('set-instructions', (instructions: MockBLPAPI.IInstruction, cb: Function): void => {
+        _.forEach(instructions, (i: any, key: string): void => {
+            MockBLPAPI.instructions[key] = i;
+        });
+        cb();
+    });
+    socket.on('clear-instructions', (arg: string[], cb: Function): void => {
+        if (arg) {
```

Comment:
This could be simplified to:
```typescript
var instructions = arg || Object.keys(MockBLPAPI.instructions);
_.forEach(instructions, (s: string): void => {
    delete MockBLPAPI.instructions[s];
}
```

Alternatively, it looks like there is no usage that uses `arg; it may be simpler to just clear everything and have the caller re-configure the instructions to execute.